lib/matplotlib/backends/backend_qt5.py

```diff
@@ -115,10 +114,8 @@ def _create_qApp():
                     is_x11_build = False
             else:
                 is_x11_build = hasattr(QtGui, "QX11Info")
-            if is_x11_build:
-                display = os.environ.get('DISPLAY')
-                if display is None or not re.search(r':\d', display):
-                    raise RuntimeError('Invalid DISPLAY variable')
+            if is_x11_build and matplotlib._c_internal_utils.invalid_display():
```

Comment:
Does this check also need to land in wx/tk/gtk flavors of the backends?

Reply:
Actually the important fix is in cbook._get_running_interactive_framework() because that's what'll determine fallback.  The one in qt5 only affects people who directly instantiate qt canvases without going through the pyplot machinery (but then it's just unlikely that they're going through this path to create their QApplication; likely they're doing it themselves and this check doesn't help).  tk and gtk throw proper exceptions when DISPLAY is invalid.  wx apparently also abort()s, but as for qt I doubt that a check would really help.